### PR TITLE
fix(receiver): traceable error wrapping

### PR DESCRIPTION
**Pull request by axelstr**
Description
-----------

Introduces error wrapping containing the called function name outside the function so that the error

```
2023-08-23T22:02:19.384Z	INFO	logger/logger.go:27	New Logger	{"level": "info"}
2023/08/23 22:02:19 listen LCM UDP multicast: setsockopt: cannot allocate memory
```

can be traced to a specific subcall.

### CODE REVIEW

File: receiver.go [hunk 1]
```diff
@@ -28,11 +28,11 @@ func ListenMulticastUDP(ctx context.Context, receiverOpts ...ReceiverOption) (*R
 	// wildcard address prefix for all administratively-scoped (local) multicast addresses
 	packetConn, err := listenConfig.ListenPacket(ctx, "udp4", fmt.Sprintf("239.0.0.0:%d", opts.port))
 	if err != nil {
-		return nil, fmt.Errorf("listen LCM UDP multicast: %w", err)
+		return nil, fmt.Errorf("listeninging packet: %w", err)
```
Comment (Edholm) [inline receiver.go:31]:
```suggestion
		return nil, fmt.Errorf("listening packet: %w", err)
```
or perhaps
```suggestion
		return nil, fmt.Errorf("opening packet listener: %w", err)
```

File: receiver.go [hunk 2]
```diff
@@ -47,13 +47,13 @@ func ListenMulticastUDP(ctx context.Context, receiverOpts ...ReceiverOption) (*R
 	if opts.interfaceName != "" {
 		ifi, err := net.InterfaceByName(opts.interfaceName)
 		if err != nil {
-			return nil, fmt.Errorf("listen LCM UDP multicast: interface %s: %w", opts.interfaceName, err)
+			return nil, fmt.Errorf("getting interface by name: interface %s: %w", opts.interfaceName, err)
```
Comment (Edholm) [inline receiver.go:50]:
```suggestion
			return nil, fmt.Errorf("getting %s interface by name: %w", opts.interfaceName, err)
```

File: receiver.go [hunk 3]
```diff
@@ -63,21 +63,21 @@ func ListenMulticastUDP(ctx context.Context, receiverOpts ...ReceiverOption) (*R
 		// Note that the service port for transport layer protocol does not matter with this operation as joining
 		// groups affects only network and link layer protocols, such as IPv4 and Ethernet.
 		if err := conn.JoinGroup(rx.ifi, &net.UDPAddr{IP: ip}); err != nil {
-			return nil, fmt.Errorf("listen LCM UDP multicast: IP %v: %w", ip, err)
+			return nil, fmt.Errorf("joining group: IP %v: %w", ip, err)
```
Comment (Edholm) [inline receiver.go:66]:
```suggestion
			return nil, fmt.Errorf("joining multicast group %v: %w", ip, err)
```

**Review by Edholm — APPROVED** [review]
beautiful and long overdue

**Review by sousandrei — APPROVED** [review]

File: receiver.go (hunk 2 above)
Comment (axelstr) [reply receiver.go]: perfect, thank you

File: receiver.go (hunk 3 above)
Comment (axelstr) [reply receiver.go]: thanks!

**Pull request merged**